test/js/substring.spec.js

```diff
@@ -0,0 +1,163 @@
+/**
+* Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
+* This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+**/
+
+const { TaintedUtils, taintFormattedString, formatTaintedValue } = require('./util')
+const assert = require('assert')
+
+const rangesTestCases = [
+  {
+    string: ':+-foobar-+:',
+    start: 0,
+    end: 3,
+    result: ':+-foo-+:'
+  },
+  {
+    string: ':+-foo-+:bar',
+    start: 3,
+    end: 6,
+    result: 'bar'
+  },
+  {
+    string: ':+-foo-+:bar',
+    start: 0,
+    end: 2,
+    result: ':+-fo-+:'
+  },
+  {
+    string: ':+-foo-+:bar',
+    start: 1,
+    end: 4,
+    result: ':+-oo-+:b'
+  },
+  {
+    string: ':+-foo-+:bar:+-baz-+:',
+    start: 3,
+    end: 8,
+    result: 'bar:+-ba-+:'
+  }
```

Comment:
Can we add more test cases?
things like that are accepted...
```
> 'abcdefg'.substring(4,2)
'cd'
```

I think that we need also test with multiple tainted ranges in the result